gems/aws-sdk-s3/spec/client_spec.rb

```diff
@@ -714,30 +714,52 @@ module S3
           part_number: 1
         }
       }.each do |operation_name, params|
-        it "handles 200 http response errors from ##{operation_name}" do
-          client.handlers
-            .remove(Seahorse::Client::Plugins::RaiseResponseErrors::Handler)
-          client.handle(step: :send) do |context|
-            context.http_response.signal_headers(200, {})
-            context.http_response.signal_data(<<-XML.strip)
-              <?xml version="1.0" encoding="UTF-8"?>
-              <Error>
-                <Code>InternalError</Code>
-                <Message>We encountered an internal error. Please try again.</Message>
-                <RequestId>656c76696e6727732072657175657374</RequestId>
-                <HostId>Uuag1LuByRx9e6j5Onimru9pO4ZVKnJ2Qz7/C1NPcfTWAtRPfTaOFg==</HostId>
-              </Error>
-            XML
-            context.http_response.signal_done
-            Seahorse::Client::Response.new(context: context)
+        describe "#{operation_name} response handling" do
+          it 'handles 200 http response errors' do
+            client.handlers.remove(
+              Seahorse::Client::Plugins::RaiseResponseErrors::Handler
+            )
+            client.handle(step: :send) do |context|
+              context.http_response.signal_headers(200, {})
+              context.http_response.signal_data(<<-XML.strip)
+                <?xml version="1.0" encoding="UTF-8"?>
+                <Error>
+                  <Code>InternalError</Code>
+                  <Message>We encountered an internal error. Please try again.</Message>
+                  <RequestId>656c76696e6727732072657175657374</RequestId>
+                  <HostId>Uuag1LuByRx9e6j5Onimru9pO4ZVKnJ2Qz7/C1NPcfTWAtRPfTaOFg==</HostId>
+                </Error>
+              XML
+              context.http_response.signal_done
+              Seahorse::Client::Response.new(context: context)
+            end
+            resp = client.send(operation_name, {
+              bucket: 'bucket',
+              key: 'key'
+            }.merge(params))
+            expect(resp.error).to be_kind_of(S3::Errors::InternalError)
+            expect(resp.context.retries).to eq(3)
+            expect(resp.data).to be(nil)
+          end
+
+          it 'handles 200 http response with empty body as error' do
+            client.handlers.remove(
+              Seahorse::Client::Plugins::RaiseResponseErrors::Handler
+            )
+            client.handle(step: :send) do |context|
+              context.http_response.signal_headers(200, {})
+              context.http_response.signal_data("\r\n")
+              context.http_response.signal_done
+              Seahorse::Client::Response.new(context: context)
+            end
+            resp = client.send(operation_name, {
+              bucket: 'bucket',
+              key: 'key'
+            }.merge(params))
+            expect(resp.error).not_to be_nil
```

Comment:
I feel like we should assert here more than not null. I would probably assert which error is being raised in case we have an error parsing which is unexpected.

Reply:
Updated to be the specific error.